terraform/modules/transfer_throttle/main.tf

```diff
@@ -0,0 +1,104 @@
+locals {
+  lambda_name = "editorial-photography-transfer_throttle-${var.environment}"
+  buckets = tomap(
+    {
+      staging = "wellcomecollection-archivematica-staging-transfer-source",
+      production = "wellcomecollection-archivematica-transfer-source"
+    }
+  )
+  target_bucket = lookup(local.buckets, var.environment)
+  input_queue_visibility_timeout = 300
+}
+
+module "input_queue" {
+  source = "../notification_queue"
+  environment = var.environment
+  queue_visibility_timeout = local.input_queue_visibility_timeout
+  action_name = "queue_shoot_transfers"
+  extra_topics = [var.upstream_topic_arn]
+}
+
+module "transfer_throttle_lambda" {
+  source = "git@github.com:wellcomecollection/terraform-aws-lambda?ref=v1.2.0"
+  name    = local.lambda_name
+  runtime = "python3.12"
+  handler = "transfer_throttle.lambda_main"
+  filename    = var.lambda_zip.output_path
+  timeout     = 300
+
+  environment = {
+    variables = {
+      SOURCE_QUEUE = module.input_queue.queue_url
+      ENVIRONMENT = var.environment
+      TARGET_TOPIC = module.output_topic.arn
+    }
+  }
+  source_code_hash = var.lambda_zip.output_base64sha256
+}
+
+
+
+module "transfer_scheduler" {
+  source = "../lambda_scheduler"
+  cron                 = "cron(30 7,9,11,13,15,16 ? * MON-FRI *)"
+  description          = "Restore a batch of shoots in the evening so they are ready to be transferred in the morning"
```

Comment:
"Moves batches of shoots to the transferrer at a rate Archivematica can handle"?

Reply:
whoops CTRL-C CTRL-V